File: src/main/scala/click/seichi/eategg/commands/Toggle.scala
package click.seichi.eategg.commands

import click.seichi.eategg.IsUuidIgnored
import org.bukkit.ChatColor
import org.bukkit.entity.Player

case class Toggle() extends Executor {
  override val commandName: String = "toggle"

  override def execute(context: CommandContext): Unit = {
    val player = context.sender match {
      case p: Player => p
      case _         =>
        context.sender.sendMessage(s"${ChatColor.RED}このコマンドはゲーム内からのみ実行できます。")
        return
    }

    if (!player.hasPermission(s"eategg.$commandName")) {
      context.sender.sendMessage(s"${ChatColor.RED}このコマンドを実行する権限がありません。")
      return
    }

    val uuid = player.getUniqueId
    IsUuidIgnored.toggle(uuid)

    val message =
      if (IsUuidIgnored.get(uuid)) s"${ChatColor.GREEN}EatEggの設定をバイパスします。"
      else s"${ChatColor.RED}EatEggの設定をバイパスしません。"
    player.sendMessage(message)
  }
}

File: src/main/java/click/seichi/eategg/commands/Toggle.java
package click.seichi.eategg.commands;

import click.seichi.eategg.IsUuidIgnored;
import org.bukkit.ChatColor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;
import org.jetbrains.annotations.NotNull;

public final class Toggle {
  public void execute(@NotNull CommandSender sender) {
    if (!(sender instanceof Player player)) {
      sender.sendMessage(ChatColor.RED + "このコマンドはゲーム内からのみ実行できます。");
      return;
    }

    if (!player.hasPermission("eategg.toggle")) {
      sender.sendMessage(ChatColor.RED + "このコマンドを実行する権限がありません。");
      return;
    }

    var uuid = player.getUniqueId();
    IsUuidIgnored.toggle(uuid);

    String message =
        IsUuidIgnored.get(uuid)
            ? ChatColor.GREEN + "EatEggの設定をバイパスします。"
            : ChatColor.RED + "EatEggの設定をバイパスしません。";
    player.sendMessage(message);
  }
}
